pennylane/circuit_graph.py

```diff
@@ -12,305 +12,278 @@
+        #nodes = [node for node in self._graph.nodes if isinstance(node, qml.operation.Observable)]
+        nodes = [node for node in self._graph.nodes if getattr(node, 'return_type', None)]
+        return sorted(nodes, key=lambda x: x.queue_idx)
 
     @property
     def operations(self):
-        """
-        Return a list of operations that do not have a return type.
-        """
-        return [node["op"] for node in self.operation_nodes]
+        """Non-observable Operations in the circuit, sorted by "idx".
 
-    @property
-    def operation_nodes(self):
-        """
-        Return a list of nodes of operations that do not have a return type, sorted by "idx".
+        Returns:
+            list[Operation]: operations
         """
-        nodes = [node for node in self.graph.nodes.values() if not node["return_type"]]
-        return sorted(nodes, key=lambda node: node["idx"])
+        #nodes = [node for node in self._graph.nodes if not isinstance(node, qml.operation.Observable)]
+        nodes = [node for node in self._graph.nodes if not getattr(node, 'return_type', None)]
+        return sorted(nodes, key=lambda x: x.queue_idx)
 
     @property
     def graph(self):
-        """The graph representation of the quantum program.
+        """The graph representation of the quantum circuit.
 
-        The resulting graph has nodes representing quantum operations,
-        and edges representing dependent/successor operations.
-
-        Each node is labelled by an integer corresponding to the position
-        in the queue; node attributes are used to store information about the node:
-
-        * ``'name'`` *(str)*: name of the quantum operation (e.g., ``'PauliZ'``)
-
-        * ``'op'`` *(Operation or Observable)*: the quantum operation/observable object
-
-        * ``'return_type'`` *(pennylane.operation.ObservableReturnTypes)*: The observable
-          return type. If an operation, the return type is simply ``None``.
+        The graph has nodes representing :class:`.Operation` instances,
+        and directed edges pointing from nodes to their immediate dependents/successors.
 
         Returns:
-            networkx.DiGraph: the directed acyclic graph representing
-            the quantum program
+            networkx.DiGraph: the directed acyclic graph representing the quantum circuit
         """
         return self._graph
 
-    def get_wire_indices(self, wire):
-        """The operation indices on the given wire.
+    def wire_indices(self, wire):
+        """Operation indices on the given wire.
 
         Args:
-            wire (int): the wire to examine
+            wire (int): wire to examine
 
         Returns:
-            list (int): all operation indices on the wire,
-            in temporal order
+            list[int]: indices of operations on the wire, in temporal order
         """
-        return [op.idx for op in self._grid[wire]]
+        return [op.queue_idx for op in self._grid[wire]]
```

Comment:
I wonder if this is still useful with the move away from operation indices. Maybe it makes sense to add a `wire_ops()` method, that returns all operations on a given wire?

Reply:
As it is, this method is only used by tests. Maybe it's not needed at all? I can remove it later if it turns out it's not needed in the QNode refactor.

Reply:
The backstory is that is was added during a previous code review as something that might be useful in future :laughing: